### ISSUE TITLE
single_char_pattern doesn't recognize scape sequences

### ISSUE THREAD

**Issue by gnzlbg** [rust-lang-nursery/rust-clippy#3075]
Example ([playground](https://play.rust-lang.org/?gist=860a7f1e0da2a5422fa5e8194f54d4cf&version=stable&mode=debug&edition=2015)):

```rust
string.find("\"");
```

errors with

```
  |
3 |     string.find("\"");
  |                        ^^^^ help: try using a char instead: `'\"'`
  |
  = note: #[warn(single_char_pattern)] on by default
  = help: for further information visit https://rust-lang-nursery.github.io/rust-clippy/v0.0.212/index.html#single_char_pattern

```

However, the appropriate char to use is `"` without the escape: `\`.

**Comment by mikerite** [comment]
The question is whether Clippy should use `'"'` over `'\"'` in suggestions.

It looks like the stdlib prefers the redundant escape form:
```
fn main() {
    println!("{:?}", '"');
    println!("{}", '"'.escape_debug());
    println!("{}", '"'.escape_default());
}
```
produces
```
'\"'
\"
\"
```

If `'"'` is better, we should also have a style lint for recommending `'\"'` to `'"'`.
